tests/orquestra/vqa/algorithms/qaoa_test.py

```diff
@@ -0,0 +1,171 @@
+    return np.random.random(hamiltonian.n_qubits)
+
+
+@pytest.fixture()
+def qaoa_object(hamiltonian, optimizer, n_layers):
+    return QAOA.default(cost_hamiltonian=hamiltonian, n_layers=n_layers)
+
+
+class TestQaoa:
+    def test_default_optimizer_init(self, hamiltonian, n_layers):
+        qaoa = QAOA.default(cost_hamiltonian=hamiltonian, n_layers=n_layers)
+        assert qaoa.optimizer.method == "L-BFGS-B"
+
+    def test_default_ansatz_init(self, hamiltonian, n_layers):
+        qaoa = QAOA.default(cost_hamiltonian=hamiltonian, n_layers=n_layers)
+        assert isinstance(qaoa.ansatz, QAOAFarhiAnsatz)
+        assert qaoa.ansatz.number_of_layers == n_layers
+
+    def test_default_estimation_init_default_value(self, hamiltonian, n_layers):
+        qaoa = QAOA.default(
+            cost_hamiltonian=hamiltonian,
+            n_layers=n_layers,
+        )
+        assert qaoa.estimation_method.__name__ == "calculate_exact_expectation_values"
+        assert qaoa._n_shots is None
+
+    def test_default_estimation_init_exact(self, hamiltonian, n_layers):
+        qaoa = QAOA.default(
+            cost_hamiltonian=hamiltonian,
+            n_layers=n_layers,
+            use_exact_expectation_values=True,
+        )
+        assert qaoa.estimation_method.__name__ == "calculate_exact_expectation_values"
+        assert qaoa._n_shots is None
+
+    def test_default_estimation_init_averaging(self, hamiltonian, n_layers):
+        n_shots = 1000
+        qaoa = QAOA.default(
+            cost_hamiltonian=hamiltonian,
+            n_layers=n_layers,
+            use_exact_expectation_values=False,
+            n_shots=n_shots,
+        )
+        assert (
+            qaoa.estimation_method.__name__
+            == "estimate_expectation_values_by_averaging"
+        )
+        assert qaoa._n_shots == n_shots
+
+    @pytest.mark.parametrize(
+        "use_exact_expectation_values,n_shots", [(True, 1000), (False, None)]
+    )
+    def test_default_raises_exception_for_invalid_inputs(
+        self, hamiltonian, n_layers, use_exact_expectation_values, n_shots
+    ):
+        with pytest.raises(ValueError):
+            _ = QAOA.default(
+                cost_hamiltonian=hamiltonian,
+                n_layers=n_layers,
+                use_exact_expectation_values=use_exact_expectation_values,
+                n_shots=n_shots,
+            )
+
+    def test_init_works(self, hamiltonian, optimizer):
+        ansatz = WarmStartQAOAAnsatz(
+            2, hamiltonian, thetas=np.random.random(hamiltonian.n_qubits)
+        )
+        estimation_method = CvarEstimator(alpha=0.5)
+        n_shots = 1000
+        qaoa = QAOA(
+            cost_hamiltonian=hamiltonian,
+            optimizer=optimizer,
+            ansatz=ansatz,
+            estimation_method=estimation_method,
+            n_shots=n_shots,
+        )
+        assert isinstance(qaoa.ansatz, WarmStartQAOAAnsatz)
+        assert isinstance(qaoa.estimation_method, CvarEstimator)
```

Comment:
We can have stronger assertions, right? We know that `qaoa.ansatz` IS precisely `ansatz`. I would advice using `is` operator in assertions in this and other similar tests.

Reply:
It actually fails when I tried to use `is`:
assert qaoa.estimation_method is CvarEstimator

Reply:
Sorry, I was a bit imprecise. What I meant is:

```python
qaoa.estimation_method is estimation_method
```

In other words, you check that the objects stored in you `QAOA` instance are the ones that you passed. 

The `qaoa.estimation_method is CvarEstimator` fails because those objects are not the same - one is actual estimator, ane the second one is a class.

Reply:
Yeah, I was not surprised when it failed, but didn't get what you mean :D 
Fixed in https://github.com/zapatacomputing/orquestra-vqa/pull/31/commits/e298363ba2fc005ccfda31412ef05fbf6783425a